File: dotnet/niuz.application.test/fixtures/InMemoryPayments.cs
using System.Collections.Generic;
using System.Linq;
using niuz.application.finance;

namespace niuz.application.fixtures
{
    public class InMemoryPayments : IPaymentRepository
    {
        readonly ICollection<Payment> payments = new List<Payment>();
        
        public void Save(Payment payment)
        {
            payments.Add(payment);
        }

        public IEnumerable<Payment> GetByBankAccount(string bankAccount)
        {
            return payments.Where(payment => payment.BankAccount == bankAccount);
        }
    }}

File: java/src/test/java/com/dddeurope/niuz/fixtures/InMemoryPayments.java
package com.dddeurope.niuz.fixtures;

import com.dddeurope.niuz.finance.Payment;
import com.dddeurope.niuz.finance.PaymentRepository;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

public class InMemoryPayments implements PaymentRepository {
    final List<Payment> payments = new ArrayList<>();

    @Override
    public void save(Payment payment) {
        payments.add(payment);
    }

    @Override
    public List<Payment> getByBankAccount(String bankAccount) {
        return payments.stream()
                       .filter(p -> Objects.equals(p.getBankAccount(), bankAccount))
                       .collect(Collectors.toList());
    }}
